packages/mask/src/plugins/Savings/SNSAdaptor/protocols/LDOProtocol.tsx

```diff
@@ -0,0 +1,102 @@
+import type Web3 from 'web3'
+import type { AbiItem } from 'web3-utils'
+import STETHABI from './steth.abi.json'
+import { EthereumTokenType } from '@masknet/web3-shared-evm'
+import type { Contract } from 'web3-eth-contract'
+import type BigNumber from 'bignumber.js'
+
+/*
+ *
+ * Other Lido Contracts can be found at
+ * https://github.com/lidofinance/lido-dao
+ *
+ */
+
+export interface LidoContract {
+    type: EthereumTokenType
+    chainName: string
+    ldoContract: string
+    stEthContract: string
+}
+
+export const LidoContracts: { [key: number]: LidoContract } = {
+    1: {
+        type: EthereumTokenType.ERC20,
+        chainName: 'Ethereum',
+        ldoContract: '0x5A98FcBEA516Cf06857215779Fd812CA3beF1B32',
+        stEthContract: '0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84',
+    },
+    5: {
+        type: EthereumTokenType.ERC20,
+        chainName: 'Gorli',
+        ldoContract: '0x56340274fB5a72af1A3C6609061c451De7961Bd4',
+        stEthContract: '0x1643E812aE58766192Cf7D2Cf9567dF2C37e9B7F',
+    },
+}
+
+export const LidoReferralAddress = `0x278D7e418a28ff763eEeDf29238CD6dfcade3A3a`
+
+export const LidoAPR = `https://cors.r2d2.to/?uri=https://stake.lido.fi/api/steth-apr`
+
+export function lidoContract(chainId: number, web3: Web3): Contract {
```

Comment:
Here I recommend constraint saving protocols with `interface`. Each protocol should new a class to implement that interface. Just copy from the lido:

```
interface SavingProtocol {
    getAPR(): Promise<string>
    getBalance(): Promise<string>
    // ... more
}

class LidoProtocol implements SavingProtocol {
    // implementation
}
```